tools/__init__.py

```diff
@@ -228,6 +229,12 @@ def _package_str(self):
             ver_str = self.package
         return ver_str
 
+    def is_attempted(self):
+        return self._is_attempted
+
+    def is_installed(self):
```

Comment:
Actually the issue is that `CondaPackage` didn't have an `is_installed` even though it should be implemented. We can always add memoization later.